Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -618,27 +702,144 @@ bool DpCatalog::allocateNode(DpBtreeNode*& newNode, const DpStateEntry& newEntry
         this->log_WARNING_HI_DpCatalogFull(newEntry.record);
         return false;
     }
+
     // get a new node from the free list
     newNode = this->m_freeListHead;
     // move the head of the free list to the next node
+    // If we've at the bottom of the free list, head will now be nullptr
     this->m_freeListHead = this->m_freeListHead->left;
 
     // initialize the new node
     newNode->left = nullptr;
     newNode->right = nullptr;
+    newNode->parent = nullptr;
     newNode->entry = newEntry;
 
     // we got one, so return success
     return true;
 }
 
-void DpCatalog::deleteEntry(DpStateEntry& entry) {}
+void DpCatalog::deallocateNode(DpBtreeNode* node) {
+    DpBtreeNode* parent = node->parent;
+
+    // since nodes are deallocated after xmit, left should be gone
+    // However, left node could be added during xmit
+    if (node->left != nullptr) {
+        // To try to only deallocate nodes at fileDone,
+        // finished nodes w/ children are trimmed out
+        // Simple shifts only work with branches that don't branch
+
+        // Since we aren't limited to adding just 1 node during file transfer
+        // left might not be a leaf
+        // Instead, find the lowest priority node on the left branch
```

Comment:
```suggestion
        // Instead, find the highest priority node on the left branch
```
I think we want the rightmost node under the left child. I'm assuming priorities go up from left to right.